### ISSUE TITLE
Error updating Kustomization status

### ISSUE THREAD

**Issue by Nicklason** [fluxcd/kustomize-controller#320]
I have tried to set up flux on my new k3s cluster using the flux bootstrap command. It is stuck at `waiting for Kustomization "flux-system/flux-system" to be reconciled` and fails because of the 5 minute timeout. The kustomize controller does apply the the kustomization, but it fails to update the status after reconcilation.

This is the error: `"Kustomization.kustomize.toolkit.fluxcd.io \"flux-system\" is invalid: status.snapshot.entries.namespace: Invalid value: \"null\": status.snapshot.entries.namespace in body must be of type string: \"null\""`

I initially thought it was a problem with v0.12.1, but then I tried v0.11.0 and got the same error.

**Comment by stefanprodan** [comment]
Did you upgraded using flux bootstrap?

**Comment by Nicklason** [comment]
I did first try to bootstrap using an existing repository that used version 0.11.0 but I had problems with that. Then I tried the bootstrap command again but this time with --version=0.12.1.

I don't remember the exact order because I tried a few things but I did try to upgrade the existing repository using the bootstrap command.

I will try reinstalling k3s and see if that fixes it.

**Comment by Nicklason** [comment]
I have set up a new cluster and used the bootstrap command with no existing GitHub repository and I got the same error.

**Comment by Nicklason** [comment]
I have managed to get it to work. I was using k3s v1.20.5 and it would fail to update the status of the kustomization, but v1.19.9 works.

**Comment by Zempashi** [comment]
Same here on a 1.20.5 cluster, with flux 0.9 and 0.12 (upgrade, and on a fresh install of the cluster and flux). For installation I just `kubectl apply -f {gotk-components.yaml,gotk-sync.yaml}\, and a bitnami's sealed-secret for the credentials. For upgrade I did
`flux install --export gotk-components.yaml && git commit && git push && kubectl apply -f gotk-components.yaml (because flux was frozen due to the bug on a dependency)`

**Comment by Zempashi** [comment]
Works on 1.21.0

**Comment by stefanprodan** [comment]
This looks like a bug in k3s, I can't replicate this on GKE/EKS nor in Kubernetes Kind that we used for e2e testing.

**Comment by Nicklason** [comment]
I would like to add that I found a workaround for my issue. Setting up flux on v1.20 k3s doesn't work, but doing it first on v1.19 k3s and then upgrading it to v1.20 works.

**Comment by stefanprodan** [comment]
Isn't there a k3s 1.21.0? Maybe that would work according to @Zempashi

**Comment by Nicklason** [comment]
I believe they are working on v1.21 for k3s but it has not been released yet.

**Comment by argamanza** [comment]
I can confirm that this bug is resolved in k3s [v1.21.0-rc1](https://github.com/k3s-io/k3s/releases/tag/v1.21.0-rc1%2Bk3s1).
Initiated a k3s cluster using
`curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" INSTALL_K3S_VERSION="v1.21.0-rc1+k3s1" sh -s -`
After a bootstrap using the same parameters that failed with v1.20.6 (see fluxcd/flux2#1344) now with the new RC version the kustomize controller finishes its update process:
```cli
{"level":"info","ts":"2021-04-26T08:09:54.015Z","logger":"controller.kustomization","msg":"Reconciliation finished in 30.720016657s, next run in 10m0s","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","name":"flux-system","namespace":"flux-system","revision":"master/f313b028f9331c23bf456407c27f1d5446f1eff7"}
{"level":"debug","ts":"2021-04-26T08:09:54.016Z","logger":"events","msg":"Normal","object":{"kind":"Kustomization","namespace":"flux-system","name":"flux-system","uid":"78bfc923-7f3e-4b2b-aa06-5e609a09962c","apiVersion":"kustomize.toolkit.fluxcd.io/v1beta1","resourceVersion":"1020"},"reason":"info","message":"Update completed"}
```
